replica/table.cc

```diff
@@ -628,12 +628,16 @@ table::update_cache(compaction_group& cg, lw_shared_ptr<memtable> m, std::vector
         }
         ms_opt = make_combined_mutation_source(std::move(sources));
     }
-    auto adder = row_cache::external_updater([this, m, ssts = std::move(ssts), new_ssts_ms = std::move(*ms_opt), &cg] () mutable {
-        for (auto& sst : ssts) {
+    auto adder = row_cache::external_updater([this, m, ssts = std::move(ssts), ms_opt = std::move(ms_opt), &cg] () mutable {
+        for (auto& i : ssts) {
+          if (auto sst = std::move(i)) {
             add_sstable(cg, sst);
             update_stats_for_new_sstable(sst);
+          }
+        }
+        if (auto new_ssts_ms = std::move(ms_opt)) {
+            m->mark_flushed(std::move(*new_ssts_ms));
```

Comment:
This violates the contract between external_updater and row_cache:

```
    // Must have strong exception guarantees. If throws, the underlying mutation source
    // must be left in the state in which it was before the call.
```

So it's not ok to partially add sstables, we should either un-add them, or abort().